pkg/cmd/pr/view/view.go

```diff
@@ -405,3 +393,46 @@ func prStateWithDraft(pr *api.PullRequest) string {
 
 	return pr.State
 }
+
+func retrievePullRequest(opts *ViewOptions) (*api.PullRequest, error) {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return nil, err
+	}
+
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	pr, repo, err := shared.PRFromArgs(apiClient, opts.BaseRepo, opts.Branch, opts.Remotes, opts.SelectorArg)
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.BrowserMode {
+		return pr, nil
+	}
+
+	errc := make(chan error)
+	count := 1
+	go func() {
+		reviews, err := api.ReviewsForPullRequest(apiClient, repo, pr)
+		pr.Reviews = *reviews
```

Comment:
Is this a safe operation to do in the goroutine? Or should `reviews` be passed back to the main thread and assigned there? Same question for `comments` below.

Reply:
In Go, we will often hear the mantra: “don't communicate by sharing memory; share memory by communicating”.

Even though your current approach should work, if we follow the above mantra, it would be generally better to return the reviews and comments from goroutines via channels and assign them to the PR in the main goroutine.

However, I can't envision a case where your current code fails, and it's relatively clean, so I would also be fine if you keep it as-is.